src/app/model/ChessEngine.ts

```diff
@@ -68,7 +70,78 @@ export class ChessEngine implements ChessModel {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
 
+  private returnWirtualProposedBoardAfterMove(chessboard: Chessboard, squareFrom: Square, squareTo: Square): Chessboard {
+    const wirtualPorposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = wirtualPorposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    return wirtualPorposedChessboard;
+  }
+
+  public static kingPosition(chessboard: Chessboard, kingSide: Side): Square | undefined {
```

Comment:
Ta metoda powinna być prywatna :) Bo nie uzywasz jej nigdzie poza klasa.

Reply:
To jeśli ma być używana poza klasa, to nie ma żadnego powodu, żeby była w klasie. TypeScript tak jak JS ma możliwość deklarowania funkcji na najwyższym poziomie. Jesli funkcja nie uzywa zadnych pol klasy to raczej powinna byc poza nia. Static stosuje sie bardzo rzadko. Raczej glownie do metod fabrykujacych. W jezykach jak Java to jest inaczej, bo tam nie ma mozliwosci funkcji bez klasy. A w tym przypadku nie wiem czy jest potrzeba wyciagac ja na zewnatrz. Wtedy tez trzeba osobno testowac i uwazac na wszelkie zmiany, zeby nie popsuc kodu zaleznego od tej funkcji. A tak to jest tylko szczegolem implementacyjnym.

Reply:
Zaraz poprawię :)